src/User32.Desktop/User32.cs

```diff
@@ -493,6 +504,68 @@ public static partial class User32
         public static extern int RegisterWindowMessage(string lpString);
 
         /// <summary>
+        /// Retrieves the name of the format from the clipboard.
+        /// </summary>
+        /// <param name="format">The type of format to be retrieved. This parameter must not specify any of the predefined clipboard formats.</param>
+        /// <param name="lpszFormatName">The format name string.</param>
+        /// <param name="nMaxCount">
+        /// The length of the <paramref name="lpszFormatName"/> buffer, in characters. The buffer must be large enough to include the terminating null character; otherwise, the format name string is truncated to <paramref name="nMaxCount"/>-1 characters.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is the number of characters copied to the buffer.
+        /// If the function fails, the return value is zero. To get extended error information, call GetLastError.
+        /// </returns>
+        [DllImport(nameof(User32), SetLastError = true, CharSet = CharSet.Auto)]
+        public static extern unsafe int GetClipboardFormatName(
+            int format,
+            [Friendly(FriendlyFlags.Array)] char* lpszFormatName,
+            int nMaxCount);
+
+        /// <summary>
+        /// Synthesizes keystrokes, mouse motions, and button clicks.
+        /// </summary>
+        /// <param name="nInputs">The number of structures in the <paramref name="pInputs" /> array.</param>
+        /// <param name="pInputs">An array of  structures. Each structure represents an event to be inserted into the keyboard or mouse input stream.</param>
+        /// <param name="cbSize">The size, in bytes, of an <see cref="INPUT" /> structure. If cbSize is not the size of an <see cref="INPUT" /> structure, the function fails.</param>
+        /// <returns>
+        /// The function returns the number of events that it successfully inserted into the keyboard or mouse input stream.
+        /// If the function returns zero, the input was already blocked by another thread. To get extended error information, call GetLastError.
+        /// </returns>
+        /// <remarks>
+        /// This function is subject to UIPI. Applications are permitted to inject input only into applications that are at an equal or lesser integrity level.
+        /// This function fails when it is blocked by UIPI. Note that neither GetLastError nor the return value will indicate the failure was caused by UIPI blocking.
+        /// </remarks>
+        [DllImport(nameof(User32), SetLastError = true)]
+        public static extern unsafe uint SendInput(
+            uint nInputs,
+            [Friendly(FriendlyFlags.Array), In] INPUT* pInputs,
```

Comment:
Can you remove the `In` attribute? I think the only time we used it we ended up removing it because it magically changed/broke the semantics of the rest of the parameters. @vbfox might recall more of the details. But we never/rarely use it, so I'd rather not use it now unless you can explain why it's needed.